Index: logger.c
===================================================================
--- logger.c.orig
+++ logger.c
@@ -27,6 +27,7 @@
 #include <time.h>
 #include <unistd.h>
 #include <netdb.h>
+#include <pthread.h>
 #include "logger.h"
 
 #define MAXLEN 128
@@ -102,17 +103,17 @@ void logger(int severity, const char *fm
      if (severity <= log_priority)
      {
           vsnprintf(message, sizeof(message), fmt, args);
-          
+
           if (log_fp)
           {
-               fprintf(log_fp, "%s %s %s[%d.%d]: %s\n", time_buf, hostname,
+               fprintf(log_fp, "%s %s %s[%d.%ld]: %s\n", time_buf, hostname,
                        log_ident, getpid(), pthread_self(), message);
                fflush(log_fp);
           }
           else if (log_syslog_is_open)
                syslog(severity, "%s", message);
           else
-               fprintf(stderr, "%s %s %s[%d.%d]: %s\n", time_buf, hostname,
+               fprintf(stderr, "%s %s %s[%d.%ld]: %s\n", time_buf, hostname,
                        log_ident, getpid(), pthread_self(), message);
      }
      va_end(args);
Index: configure.ac
===================================================================
--- configure.ac.orig
+++ configure.ac
@@ -66,7 +66,7 @@ AC_PROG_CC
 dnl Check for AIX
 AC_AIX
 
-CFLAGS="-g -Wall -D_REENTRANT"
+CFLAGS="${CFLAGS} -g -Wall -D_REENTRANT"
 
 if test x$debug = xtrue; then
    CFLAGS="$CFLAGS -O0 -DDEBUG"
Index: tftpd_list.c
===================================================================
--- tftpd_list.c.orig
+++ tftpd_list.c
@@ -143,13 +143,13 @@ int tftpd_list_find_multicast_server_and
      char options[MAXLEN];
      char string[MAXLEN];
      char *index;
-     int len;
+     size_t len;
 
      *thread = NULL;
 
      opt_request_to_string(tftp_options, options, MAXLEN);
      index = strstr(options, "multicast");
-     len = (int)index - (int)options;
+     len = (size_t)(index - options);
 
      /* lock the whole list before walking it */
      pthread_mutex_lock(&thread_list_mutex);
@@ -166,7 +166,7 @@ int tftpd_list_find_multicast_server_and
                     /* must have exact same option string */
                     if (strncmp(string, options, len) == 0)
                     {
-                         *thread = current;                         
+                         *thread = current;
                          /* insert the new client at the end. If the client is already
                             in the list, don't add it again. */
                          tmp = current->client_info;
@@ -188,17 +188,17 @@ int tftpd_list_find_multicast_server_and
                          }
                          tmp->next = client;
                          /* unlock mutex and exit */
-                         pthread_mutex_unlock(&current->client_mutex);                    
+                         pthread_mutex_unlock(&current->client_mutex);
                          pthread_mutex_unlock(&thread_list_mutex);
                          return 1;
                     }
                }
-               pthread_mutex_unlock(&current->client_mutex);                    
+               pthread_mutex_unlock(&current->client_mutex);
           }
           current = current->next;
      }
      pthread_mutex_unlock(&thread_list_mutex);
-     
+
      return 0;
 }
 
Index: tftp_mtftp.c
===================================================================
--- tftp_mtftp.c.orig
+++ tftp_mtftp.c
@@ -63,7 +63,7 @@ extern int tftp_cancel;
  * If mode = 0, count missed packet from block 0. Else, start after first
  * received block.
  */
-int tftp_mtftp_missed_packet(int file_bitmap[], int last_block, int mode)
+int tftp_mtftp_missed_packet(unsigned int file_bitmap[], int last_block, int mode)
 {
      int missed_block = 0;
      int block_number = 0;
